### ISSUE TITLE
Showing folder size in details view

### ISSUE THREAD

**Issue by pbakes** [files-community/Files#3944]
There are so many third-party file explorer's that offer showing folder size, e.g. Explorer++, Xinorbis, WizTree, TreeSize, still nothing that allows you to view folder size in Explorer. Files, as far as the UI is concerned, is much better than all those alternatives. It would be nice to see that functionality implemented.

![12059](https://user-images.githubusercontent.com/38716120/110679283-24ef5000-818c-11eb-923a-915abbcdf027.jpg)

**Comment by gave92** [comment]
This is something we have looked at in the past and decided against implementing due to the performance issues involved.
Not sure if the new caching feature could change this.

**Comment by d2dyno1** [comment]
Perhaps a setting for this would suffice?

**Comment by R3voA3** [comment]
It's important that the preview is not slowed down any further, it's already slow for folder with many child folders.

**Comment by yaichenbaum** [comment]
@gave92 Do you think that if we added a calculate button, that it should calculate all items, or would it be an individual option for each folder?

**Comment by gave92** [comment]
@yair100 I was thinking about a button that calculates size for all currently selected folders.

**Comment by gave92** [comment]
Looking around this should be the fastest way to get folder size (from fulltrustprocess):
```cs
Type fsoType = Type.GetTypeFromProgID("Scripting.FileSystemObject");
dynamic fso = Activator.CreateInstance(fsoType);
var folder = fso.GetFolder(@"C:\Users\Marco\source\repos\Files");
Console.WriteLine(folder.Size);
System.Runtime.InteropServices.Marshal.ReleaseComObject(folder);
System.Runtime.InteropServices.Marshal.ReleaseComObject(fso);
```

**Comment by cinqmilleans** [comment]
@gave92 @yair100 @d2dyno1
I did a trial implementation. This does not reduce performance so much because the sizes are calculated after displaying the rest. I do think, however, that this should be a setting, (disabled by default). The implementation only calculates once per element and displays in the size column and footer.

https://user-images.githubusercontent.com/46631671/144085880-1106214d-1805-4555-ad85-1e4e07f5b0e4.mp4

**Comment by yaichenbaum** [comment]
@cinqmilleans that doesn't look too bad, does it have any impact on performance at all?

**Comment by cinqmilleans** [comment]
@yair100 I did not see any noticeable loss. The sizes update last without blocking the user interface. We can therefore do everything even if the sizes have not finished loading: open a sub-folder, rename, ....

**Comment by yair100** [comment]
> @yair100 I did not see any noticeable loss. The sizes update last without blocking the user interface. We can therefore do everything even if the sizes have not finished loading: open a sub-folder, rename, ....

@cinqmilleans do you want to add this as an experimental option?

**Comment by cinqmilleans** [comment]
@yair100 Yes of course. I prepare this.

**Comment by yaichenbaum** [comment]
@pbakes this issue should be resolved with #7244.
